src/ServiceFabric/ServiceFabric/Commands/ManagedClusters/ManagedClusters/AddAzServiceFabricManagedClusterNetworkSecurityRule.cs

```diff
@@ -0,0 +1,163 @@
+
+namespace Microsoft.Azure.Commands.ServiceFabric.Commands
+{
+	[Cmdlet(VerbsCommon.Add, ResourceManager.Common.AzureRMConstants.AzurePrefix + Constants.ServiceFabricPrefix + "ManagedNetworkSecurityRule", DefaultParameterSetName = ByObj, SupportsShouldProcess = true), OutputType(typeof(PSManagedCluster))]
+	public class AddAzServiceFabricManagedClusterNetworkSecurityRule : ServiceFabricManagedCmdletBase
+	{
+		protected const string ByName = "ByName";
+		protected const string ByObj = "ByObj";
+
+		#region Params
+
+		#region Common params
+
+		[Parameter(Mandatory = true, Position = 0, ValueFromPipelineByPropertyName = true, ParameterSetName = ByName,
+			HelpMessage = "Specify the name of the resource group.")]
+		[ResourceGroupCompleter]
+		[ValidateNotNullOrEmpty()]
+		public string ResourceGroupName { get; set; }
+
+		[Parameter(Mandatory = true, Position = 1, ValueFromPipelineByPropertyName = true, ParameterSetName = ByName,
+			HelpMessage = "Specify the name of the cluster.")]
+		[ResourceNameCompleter(Constants.ManagedClustersFullType, nameof(ResourceGroupName))]
+		[ValidateNotNullOrEmpty()]
+		public string ClusterName { get; set; }
+
+		[Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = ByObj,
+			HelpMessage = "Cluster resource")]
+		[ValidateNotNull]
+		public PSManagedCluster InputObject { get; set; }
+
+		#endregion
+
+		[Parameter(Mandatory = true, HelpMessage = "Gets or sets the network traffic is allowed or denied. Possible values include: 'allow', 'deny'")]
+		public string Access { get; set; }
+
+		[Parameter(Mandatory = false, HelpMessage = "Gets or sets network security rule description.")]
+		public string Description { get; set; }
+
+		[Parameter(Mandatory = true, HelpMessage = "Gets or sets the destination address prefixes. CIDR or destination IP ranges.")]
+		public string[] DestinationAddressPrefixes { get; set; }
+
+		[Parameter(Mandatory = true, HelpMessage = "Gets or sets the destination port ranges.")]
+		public string[] DestinationPortRanges { get; set; }
+
+		[Parameter(Mandatory = true, HelpMessage = "Gets or sets network security rule direction. Possible values include: 'inbound', 'outbound'")]
+		public string Direction { get; set; }
+
+		[Parameter(Mandatory = true, HelpMessage = "network security rule name.")]
+		[Alias("NetworkSecurityRuleName")]
+		public string Name { get; set; }
+
+		[Parameter(Mandatory = true, HelpMessage = "Gets or sets the priority of the rule. The value can be in the range 1000 to 3000. Values outside this range are reserved for Service Fabric ManagerCluster Resource Provider. The priority number must be unique for each rule in the collection. The lower the priority number, the higher the priority of the rule.")]
+		public int Priority { get; set; }
+
+		[Parameter(Mandatory = true, HelpMessage = "Gets or sets network protocol this rule applies to. Possible values include: 'http', 'https', 'tcp', 'udp', 'icmp', 'ah', 'esp'")]
```

Comment:
should add to helpmessage it can also be `any`

Reply:
I tested this and it failed.

Reply:
Also for some reason all the properties were required except for 'description' otherwise the deployment doesn't work. So I made them mandatory

Reply:
https://learn.microsoft.com/en-us/azure/templates/microsoft.network/networksecuritygroups/securityrules?pivots=deployment-language-arm-template#securityrulepropertiesformat-1 looking at the swagger docs `Any == *` so add that `*` works to support any and test that out. 

That is interesting you found all but description are required because the swagger docs only claim that name protocol access and direction are required

Reply:
@a-santamaria  how should I handle this ?

Reply:
@mwesigwaguma you need to add an asterisk as an option for this - not any. As in the docs above the supported value for any protocol is `*`